src/DataTypes/DataTypeNumberBase.h

```diff
@@ -20,11 +20,14 @@ class DataTypeNumberBase : public IDataType
     static constexpr bool is_parametric = false;
     static constexpr auto family_name = TypeName<T>;
     static constexpr auto type_id = TypeToTypeIndex<T>;
+    // Create a map from the name of the type to the name of the type in MySQL.
+    static const std::map<std::string, std::string> mysqlTypeMap;
 
     using FieldType = T;
     using ColumnType = ColumnVector<T>;
 
     const char * getFamilyName() const override { return TypeName<T>.data(); }
+    const char * getSQLCompatibleName() const override { return mysqlTypeMap.at(TypeName<T>.data()).c_str(); }
```

Comment:
We should avoid the usage of TypeName. I think writing a series of `if constexpr` directly here will be better.